### PR TITLE
[CT-1216] Deliver all metrics feature.

### PR DESCRIPTION
**Pull request by CrazyPit**
[CT-1256](https://toptal-core.atlassian.net/browse/CT-1256)

Some project (e.g. TopScheduler) does not need a feature with enabling/disabling delivery for specify metrics. 
Now you can pass `default_delivery = true` or even `false` and make delivering default for all metrics enabled or disabled.

### CODE REVIEW

File: lib/monitoring_sdk/metrics.rb [hunk 1]
```diff
@@ -52,8 +52,14 @@ class Metrics
...
     #   (e.g. `MonitoringSDK::MetricsPublishAdapters::ToptalWaterDrop`) with publish method receiving message hash.
-    # @param delivered_metrics [Hash|Config::Options] - configuration for metric that should be delivered in the
+    # @param delivered_metrics [Hash|Config::Options|nil] - configuration for metric that should be delivered in the
     #   current environment. For many reasons we need to disable/enable some metrics only in concrete environments.
+    #   You should not use `delivered_metrics` along with `default_delivery`.
+    # @param default_delivery [Boolean|nil]
+    #   If you project does not need `delivered_metrics` feature you can use `default_delivery`.
+    #   In case it is set to true all metrics will be delivered.
+    #   In case it is set to false none metrics will be delivered.
+    #   You should not use `delivered_metrics` along with `default_delivery`.
```
Comment (ojab) [inline lib/monitoring_sdk/metrics.rb:62]: hmm, why not both? I. e. `default_delivery: true && delivered_metrics: false` do not specific deliver metric and vice a versa.

Comment (CrazyPit) [reply lib/monitoring_sdk/metrics.rb]: Yeah, you're right, fixed.

File: spec/monitoring_sdk/metrics_spec.rb [hunk 2]
```diff
@@ -102,5 +104,29 @@
...
+      let(:default_delivery) { true }
+
+      include_examples 'does not delivers metrics'
+    end
+
+    context 'when default_delivery is false' do
+      let(:delivered_metrics) { {} }
+      let(:default_delivery) { false }
+
+      include_examples 'does not delivers metrics'
```
Comment (ojab) [inline spec/monitoring_sdk/metrics_spec.rb:129]: `when default_delivery is false but it is true for specific metric` is missing, otherwise LGTM.

Comment (CrazyPit) [reply spec/monitoring_sdk/metrics_spec.rb:129]: this is a default variant, tested in the first spec.

**Pull request merged**